### PR TITLE
Create indicator runner main function

### PR DESCRIPTION
**Pull request by sgsmob**
### Description
Creates a main function that runs an arbitrary indicator.

### Changelog
- `runner.py`:  Adds a main function that can be invoked with `python -m delphi_utils.runner [indicator module]`
- `archive.py`:  Refactor to have an archiver creator function

### CODE REVIEW

File: _delphi_utils_python/delphi_utils/runner.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
 """Indicator running utilities."""
+import argparse as ap
+import importlib
 from typing import Any, Callable, Dict, Optional
 from .archive import ArchiveDiffer
+from .archive import from_params as archiver_from_params
```
Comment (chinandrew) [inline _delphi_utils_python/delphi_utils/runner.py:6]: why not name each function `archiver/validator_from_params` in the definition? is it so there's a common function name across each of the objects?

File: _delphi_utils_python/delphi_utils/archive.py [hunk 2]
```diff
@@ -621,46 +621,13 @@ def update_cache(self):
...
-               common_params["export_dir"],
-               aws_credentials=archive_params.get("aws_credentials", {}),
-               branch_name=args.branch_name,
-               bucket_name=archive_params.get("bucket_name", ""),
-               commit_message=args.commit_message,
-               commit_partial_success=args.commit_partial_success,
-               indicator_prefix=args.indicator_prefix,
-               override_dirty=args.override_dirty
-               )
+    if "archive" not in _params:
```
Comment (chinandrew) [inline _delphi_utils_python/delphi_utils/archive.py:630]: what's this doing? if this evals to true, shouldn't the archiver not run?

File: _delphi_utils_python/delphi_utils/archive.py [hunk 3]
```diff
@@ -98,43 +97,44 @@ def diff_export_csv(
...
-        arch_diff = FilesystemArchiveDiffer(cache_dir,
-                                            export_dir)
-    else:
-        raise ValueError(f"No archive type named '{archive_type}'")
-    arch_diff.run()
+    if "archive" not in params:
+        return None
+
+    common_params = params["common"]
+    export_dir = common_params["export_dir"]
```
Comment (chinandrew) [inline _delphi_utils_python/delphi_utils/archive.py:119]:
```suggestion
    export_dir = common_params["common"]["export_dir"]
```

Don't feel strongly about this recommendation, seems like extracting the sub-dicts to their own variable is done a few times here. Also looks like this is mainly copy paste.

File: _delphi_utils_python/delphi_utils/archive.py (hunk 2 above)
Comment (sgsmob) [reply _delphi_utils_python/delphi_utils/archive.py:630]: The thought here is that if you are explicitly calling the module from the command line, you intend to run archiving.  If we can't find "archive" in the params, we are interpreting that as "this params.json has not been refactored to use the new style" (see #847 and the old version of this).

Comment (chinandrew) [reply _delphi_utils_python/delphi_utils/archive.py:630]: ah got it, thanks. can you leave a comment explaining the old structure bit?

File: _delphi_utils_python/delphi_utils/runner.py (hunk 1 above)
Comment (sgsmob) [reply _delphi_utils_python/delphi_utils/runner.py]: Done

File: _delphi_utils_python/delphi_utils/archive.py (hunk 3 above)
Comment (sgsmob) [reply _delphi_utils_python/delphi_utils/archive.py:119]: You are actually suggesting to change this to `export_dir = params["common"]["export_dir"]`, yes?

File: _delphi_utils_python/delphi_utils/archive.py (hunk 2 above)
Comment (sgsmob) [reply _delphi_utils_python/delphi_utils/archive.py:634]: Done

File: _delphi_utils_python/delphi_utils/archive.py (hunk 3 above)
Comment (chinandrew) [reply _delphi_utils_python/delphi_utils/archive.py:119]: yea

File: _delphi_utils_python/delphi_utils/archive.py [hunk 4]
```diff
@@ -98,43 +97,44 @@ def diff_export_csv(
...
-        The directory with most recent exported CSVs to diff to.
-    **kwargs:
-        Keyword arguments corresponding to constructor arguments for the respective ArchiveDiffers.
+    params: Dict[str, Dict[str, Any]]
+        Dictionary of user-defined parameters with the following structure:
+        - "common":
+            - "export_dir": str, directory to which indicator output files have been exported
+        - "archive":
+            - "cache_dir": str, directory containing cached data from previous indicator runs
+            - "branch_name" (required for git archiver): str, name of git branch
```
[the remark this answers is not in the document]
Comment (sgsmob) [reply _delphi_utils_python/delphi_utils/archive.py:113]: Done

File: _delphi_utils_python/delphi_utils/archive.py (hunk 3 above)
Comment (sgsmob) [reply _delphi_utils_python/delphi_utils/archive.py]: This is obsolete now that I have changed the logic

**Comment by sgsmob** [comment]
I made a bunch of changes to the logic that decides what archiver to use to the point that it needed associated tests.  Please take a look now.

**Comment by chinandrew** [comment]
Looks like you hit #896, restarted tests and they passed now

**Review by chinandrew — APPROVED** [review]
:shipit::shipit::shipit:

**Pull request merged**